packages/reactive_ble_mobile/ios/Classes/Plugin/PluginController.swift

```diff
@@ -491,6 +497,66 @@ final class PluginController {
         completion(.success(result))
     }
 
+    func writeDescriptorWithoutResponse(name: String, args: WriteDescriptorRequest, completion: @escaping PlatformMethodCompletionHandler) {
+        guard let central = central
+        else {
+            completion(.failure(PluginError.notInitialized.asFlutterError))
+            return
+        }
+        
+        guard let descriptor = QualifiedDescriptorIDFactory().make(from: args.descriptor)
+        else {
+            completion(.failure(PluginError.invalidMethodCall(method: name, details: "descriptor, characteristic, service, and peripheral IDs are required").asFlutterError))
+            return
+        }
+        
+        guard let peripheral = central.activePeripherals[descriptor.peripheralID],
+              let service = peripheral.services?.first(where: { service in
```

Comment:
how will `.first` work with devices that provide several instances of the same service (UUID)?  Same for several instances of the same characteristic UUID.

Reply:
Devices can provide multiple services with the same UUID? I have to admit that's a part of the spec I do not know about!

Technically this May be a more fundamental problem in this package then, looking as QualifiedCharacteristic for example, a full lookup is done using the UUIDs but not via index?

Reply:
In case anyone wants to continue with this in the future. Have a look at https://github.com/PhilipsHue/flutter_reactive_ble/pull/776 where the problem of multiple characteristics of services with the same id has been addressed. Something similar would be needed here.